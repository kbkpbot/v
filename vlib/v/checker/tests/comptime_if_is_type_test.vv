module main

fn foo[T](val T) {
	$if T is bool || T is $int {
		println('bool || \$int')
	} $else {
		$compile_error('only support bool and num type')
	}


}

fn test_main() {
	foo(123)
}
