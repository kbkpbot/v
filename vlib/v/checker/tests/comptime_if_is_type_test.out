comptime_if_is_type_test.vv:7:3: cgen error: only support bool and num type
    5 |         println('bool || \$int')
    6 |     } $else {
    7 |         $compile_error('only support bool and num type')
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    8 |     }
    9 |
